### PR TITLE
Add pandas-stubs for mypy and fix uncovered typing issues

### PR DESCRIPTION
**Pull request by gsakkis**
Add public type stubs for `pandas` via [pandas-stubs](https://pypi.org/project/pandas-stubs/) and fix the 9 resulting mypy errors.

### CODE REVIEW

File: apis/python/src/tiledbsoma/types.py [hunk 1]
```diff
@@ -11,10 +11,12 @@
     NPInteger = np.integer[npt.NBitBase]
     NPFloating = np.floating[npt.NBitBase]
     NDArray = npt.NDArray[np.number[npt.NBitBase]]
```
Comment (johnkerl) [inline apis/python/src/tiledbsoma/types.py:13]: While we're here this needs to be `NPNDArray`

We already have `SparseNDArray` and `DenseNDArray` for SOMA, and this really needs the `NP` prefix like the other NumPy types already have, and as the Pandas types have the `PD` prefix

**Review by johnkerl — CHANGES_REQUESTED** [review]
This is awesome @gsakkis -- thanks!!

One requested change which I think is important.

If you don't want to do it on this PR I'll do it on another.

File: apis/python/src/tiledbsoma/types.py (hunk 1 above)
Comment (gsakkis) [reply apis/python/src/tiledbsoma/types.py]: Makes sense; renamed it [here](https://github.com/single-cell-data/TileDB-SOMA/pull/772/commits/cd21d417d4c2bdc42f3263c959dfa2a1735446d4) :+1:

**Review by johnkerl — APPROVED** [review]
🚢

**Pull request merged**